xbmc/guilib/GUITexture.cpp

```diff
@@ -27,6 +30,27 @@ CTextureInfo::CTextureInfo(const std::string &file):
   useLarge = false;
 }
 
+void CGUITexture::Register(const CreateGUITextureFunc& createFunction,
+                           const DrawQuadFunc& drawQuadFunction)
+{
+  m_createGUITextureFunc = createFunction;
+  m_drawQuadFunc = drawQuadFunction;
+}
+
+CGUITexture* CGUITexture::CreateTexture(
+    float posX, float posY, float width, float height, const CTextureInfo& texture)
+{
+  return m_createGUITextureFunc(posX, posY, width, height, texture);
```

Comment:
You should handle the case where `m_createGUITextureFunc` is unregistered.

Reply:
Thanks for catching that. I added a check if the method has been registered.